crates/router/src/core/payment_link/payment_link_initiate/payment_link.js

```diff
@@ -617,26 +617,54 @@ function renderDynamicMerchantDetails(paymentDetails) {
 }
 
 function appendMerchantDetails(paymentDetails, merchantDynamicDetails) {
-  if (!(typeof paymentDetails.transaction_details === "string" && paymentDetails.transaction_details.length > 0)) {
+  if (
+    !(
+      typeof paymentDetails.transaction_details === "object" &&
+      Object.keys(paymentDetails.transaction_details).length > 0
+    )
+  ) {
     return;
   }
 
   try {
-    let merchantDetailsObject = JSON.parse(paymentDetails.transaction_details);
+    let merchantDetailsObject = paymentDetails.transaction_details;
+    // sort the merchant details based on the position
+    // if position is null, then it will be shown at the end
+    merchantDetailsObject.sort((a, b) => {
+      if (a.ui_configuration === null || a.ui_configuration.position === null)
+        return 1;
+      if (b.ui_configuration === null || b.ui_configuration.position === null)
+        return -1;
+
+      return a.ui_configuration.position - b.ui_configuration.position;
```

Comment:
position can be optional (defined in `TransactionDetailsUiConfiguration`), we can add type checks for the behaviour to be consistent